lib/shopify_app/controller_concerns/token_exchange.rb

```diff
@@ -83,8 +89,36 @@ def redirect_to_bounce_page
       )
     end
 
+    def missing_embedded_param?
+      !params[:embedded].present? || params[:embedded] != "1"
+    end
+
+    def redirect_to_embed_app
+      ShopifyApp::Logger.debug("Redirecting to embed app")
+
+      host = if params[:host]
+        params[:host]
+      elsif params[:shop]
+        Base64.encode64("#{sanitized_shop_name}/admin")
```

Comment:
This is a little bit of an assumption - I think there are some cases where we might not want this (requests aren't coming from the admin), but we can deal with those if / when they happen.

Reply:
There are cases where `host` isn't necessarily `shop/admin`, so we're basically assuming that will be the case whenever we hit this path.